### PR TITLE
Solve well equation initially_continuing from PR OPM/opm-autodiff#396

### PR DESCRIPTION
**Pull request by GitPaean**
This PR is the updated version of the PR OPM/opm-autodiff#396 with updating most of the comments there. The new PR is producing exactly the same behavior with the original PR when running the Norne case. 

Some warnings are generated compared with the master branch, which should also be fixed before merging.

### CODE REVIEW

File: opm/autodiff/BlackoilModelBase_impl.hpp [hunk 1]
```diff
@@ -1124,6 +1217,91 @@ namespace detail {
...
+            }
+
+            ++it;
+            std::vector<ADB> eqs;
+            eqs.reserve(2);
+            eqs.push_back(residual_.well_flux_eq);
+            eqs.push_back(residual_.well_eq);
+            ADB total_residual = vertcatCollapseJacs(eqs);
+            const std::vector<M>& Jn = total_residual.derivative();
+            const Eigen::SparseLU< M > solver(Jn[0]);
```
Comment (GitPaean) [inline opm/autodiff/BlackoilModelBase_impl.hpp]: This line causes a warning. Please suggest how to handle it. Thanks. 
```
/usr/local/include/eigen3/Eigen/src/OrderingMethods/Ordering.h:135:13: warning: variable ‘info’ set but not used [-Wunused-but-set-variable]
```

Comment (bska) [inline opm/autodiff/BlackoilModelBase_impl.hpp]: The `Ordering.h` header file is part of Eigen so there's no way of fixing that--short of patching your Eigen installation of course.

Comment (atgeirr) [inline opm/autodiff/BlackoilModelBase_impl.hpp]: Adding this warning to the ignored list in opm/core/utility/platform_dependent/disable_warnings.h would suppress it for those who have turned on SILENCE_EXTERNAL_WARNINGS.

Comment (bska) [inline opm/autodiff/BlackoilModelBase_impl.hpp]: > Adding this warning to the ignored list [...] would suppress it

First, that's a **really** bad idea.  We want to be aware of "set-but-not-used".  Second, I don't see how the suppression mechanism can possibly work in this case.  The context of `info` (at least in Eigen 3.2.1) is
```C++
template<typename Index>
class COLAMDOrdering
{
  public:
    typedef PermutationMatrix<Dynamic, Dynamic, Index> PermutationType;
    typedef Matrix<Index, Dynamic, 1> IndexVector;

    /** Compute the permutation vector form a sparse matrix */
    template <typename MatrixType>
    void operator() (const MatrixType& mat, PermutationType& perm)
    {
      Index m = mat.rows();
      Index n = mat.cols();
      Index nnz = mat.nonZeros();
      // Get the recommended value of Alen to be used by colamd
      Index Alen = internal::colamd_recommended(nnz, m, n);
      // Set the default parameters
      double knobs [COLAMD_KNOBS];
      Index stats [COLAMD_STATS];
      internal::colamd_set_defaults(knobs);

      Index info;
      IndexVector p(n+1), A(Alen);
      for(Index i=0; i <= n; i++)   p(i) = mat.outerIndexPtr()[i];
      for(Index i=0; i < nnz; i++)  A(i) = mat.innerIndexPtr()[i];
      // Call Colamd routine to compute the ordering 
      info = internal::colamd(m, n, Alen, A.data(), p.data(), knobs, stats);
      eigen_assert( info && "COLAMD failed " );

      perm.resize(n);
      for (Index i = 0; i < n; i++) perm.indices()(p(i)) = i;
    }
};
```
so the class is a template.  In other words, we would in general have to guard every call site to suppress the warning.

Comment (atgeirr) [inline opm/autodiff/BlackoilModelBase_impl.hpp]: We want to be aware of set-but-not-used in *our* code, but as usual I think it should be easy to ignore third-party warnings (and no-one is forced to use that option). The suppression mechanism seems to work well, even for a template, which I agree was a pleasant little surprise.

**Comment by atgeirr** [comment]
Thanks to @totto82 for the patch, and to @GitPaean for the fixup. Will merge.

**Pull request merged**